ConfigSpace/util.pyx

```diff
@@ -436,3 +436,185 @@ def fix_types(configuration: dict,
+
+        Parameters
+        ----------
+        num_steps_dict: dict
+            Same description as above
+
+        hp_name: str
+            Hyperparameter name
+
+        Returns
+        -------
+        tuple
+            Holds grid values for the given hyperparameter
+
+        '''
+        param = configuration_space.get_hyperparameter(hp_name)
+        if isinstance(param, (CategoricalHyperparameter)):
+            return param.choices
+
+        elif isinstance(param, (OrdinalHyperparameter)):
+            return param.sequence
+
+        elif isinstance(param, Constant):
+            return tuple([param.value, ])
+
+        elif isinstance(param, UniformFloatHyperparameter):
+            if param.log:
+                lower, upper = np.log([param.lower, param.upper])
+            else:
+                lower, upper = param.lower, param.upper
+
+            if param.name in num_steps_dict:
+                num_steps = num_steps_dict[param.name]
+                grid_points = np.linspace(lower, upper, num_steps)
+            else:
+                grid_points = np.arange(lower, upper + param.q, param.q) # check for log and for rounding issues
+
+            if param.log:
+                grid_points = np.exp(grid_points)
+
+            # Avoiding rounding off issues
+            print(grid_points)
+            if grid_points[0] < param.lower:
+                grid_points[0] = param.lower
+            if grid_points[-1] > param.upper:
+                grid_points[-1] = param.upper
+
+            return tuple(grid_points)
+
+        elif isinstance(param, UniformIntegerHyperparameter):
+            if param.log:
+                lower, upper = np.log([param.lower, param.upper])
+            else:
+                lower, upper = param.lower, param.upper
+
+            if param.name in num_steps_dict:
+                num_steps = num_steps_dict[param.name]
+                grid_points = np.linspace(lower, upper, num_steps)
+            else:
+                grid_points = np.arange(lower, upper + param.q, param.q) # check for log and for rounding issues
+
+            if param.log:
+                grid_points = np.exp(grid_points)
+            grid_points = grid_points.astype(int)
+
+            # Avoiding rounding off issues
+            if grid_points[0] < param.lower:
+                grid_points[0] = param.lower
+            if grid_points[-1] > param.upper:
+                grid_points[-1] = param.upper
+
+            return tuple(grid_points)
+
+        else:
+            raise TypeError("Unknown hyperparameter type %s" % type(param))
+
+    def get_cartesian_product(value_sets, hp_names):

```

Comment:
Same as above, could you please add type annotations here?

Reply:
Done